### ISSUE TITLE
conditionalAssignment matches expression within try block

### ISSUE THREAD

**Issue by antiraum** [nicklockwood/SwiftFormat#1536]
Version 0.52.6 / Swift 5.9

`conditionalAssignment` reformats the following code

```
        let user2: User? = try {
            if let data2 = defaults.data(forKey: defaultsKey) {
                return try PropertyListDecoder().decode(User.self, from: data2)
            } else {
                return nil
            }
        }()
```

to

```
        let user2: User? = try if let data2 = defaults.data(forKey: defaultsKey) {
            try PropertyListDecoder().decode(User.self, from: data2)
        } else {
            nil
        }
```

Which doesn't seem to be allowed. The Xcode error message is

> 'try' may not be used on 'if' expression

**Comment by nicklockwood** [comment]
Paging @calda

**Comment by nicklockwood** [comment]
@calda do you think it's sufficient to remove the stray `try` here (and maybe do the same for `await`?) or would it be safer just to not remove the closure at all in this scenario?

**Comment by nicklockwood** [comment]
@antiraum FYI I think it's actually the `redundantClosure` rule that's a fault here.

**Comment by calda** [comment]
Ah, yes, this looks like a bug in `redundantClosure`. We'd need to either drop the proceeding `try` before the closure call if present.

**Comment by calda** [comment]
I'll post a fix.

**Comment by nicklockwood** [comment]
@calda cool, thanks for confirming - I'll take care of it.

**Comment by nicklockwood** [comment]
@calda fix here: https://github.com/nicklockwood/SwiftFormat/commit/59e0fb032bfa6f783d268b4cee766a7b5a5e5f9e

**Comment by nicklockwood** [comment]
@antiraum fixed in 0.52.7

**Comment by antiraum** [comment]
Thanks a lot!
